packages/http-client-java/generator/http-client-generator-core/src/main/java/com/microsoft/typespec/http/client/generator/core/mapper/ModelPropertyMapper.java

```diff
@@ -203,7 +203,9 @@ public ClientModelProperty map(Property property, boolean mutableAsOptional) {
 
         if (property.getSchema() instanceof ConstantSchema) {
             Object objValue = ((ConstantSchema) property.getSchema()).getValue().getValue();
-            builder.constant(true);
+            // typically "optional-constant-as-enum=true" for DPG/MGMT
+            // but it could be "false" in vanilla, in which case, the property can be set to null
+            builder.constant(property.isRequired());
```

Comment:
Major change here, as commented.

It is a pretty corner case in vanilla, that a constant being optional (hence it can be set to `null` -- not really a constant).